drivers/adc/adc_mcux_lpadc.c

```diff
@@ -496,7 +496,9 @@ static int mcux_lpadc_init(const struct device *dev)
 	adc_config.conversionAverageMode = config->calibration_average;
 #endif /* FSL_FEATURE_LPADC_HAS_CTRL_CAL_AVGS */
 
-	adc_config.powerLevelMode = config->power_level;
+#if !(DT_ANY_INST_HAS_PROP_STATUS_OKAY(no_power_level))
```

Comment:
last comment is that this doesn't have to be a preprocessor #if, it can be a C if(DT_ANY_INST_etc(..)), this will help keep the code more testable since the file content doesn't change as much between different configurations